### PR TITLE
feat: Conversation MLS verification status updating (WPB-3872)

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3872" title="WPB-3872" target="_blank">StoryWPB-3872</a>  [Android] Create separate unverified/verified/degraded state machine for MLS/E2EI
  </td></table>
  <br />
 

# What's new in this PR?

In fact just adopting AR to changes from [kalium PR](https://github.com/wireapp/kalium/pull/2035) 
including: 
removing usage of `GetConversationVerificationStatusUseCase` as the observing of conversation verification state is done in `ObserveConversationDetailsUseCase`

Needs releases with:

- https://github.com/wireapp/kalium/pull/2035

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [715](https://10.10.124.134/job/AR-build-pipeline/715/) **failed**.

**Comment by AndroidBob** [comment]
Build [727](https://10.10.124.134/job/AR-build-pipeline/727/) **failed**.

**Comment by AndroidBob** [comment]
Build [734](https://10.10.124.134/job/AR-build-pipeline/734/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7557142)(7557142)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7557142)(7557142)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [764](https://10.10.124.134/job/AR-build-pipeline/764/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7564209)(7564209)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7564209)(7564209)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [767](https://10.10.124.134/job/AR-build-pipeline/767/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7565438)(7565438)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7565438)(7565438)-staging-compat.apk)

**Comment by vitorhugods** [comment]
Just converting to draft while there is no description.

**Comment by AndroidBob** [comment]
Build [790](https://10.10.124.134/job/AR-build-pipeline/790/) **failed**.

**Comment by AndroidBob** [comment]
Build [794](https://10.10.124.134/job/AR-build-pipeline/794/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7573394)(7573394)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7573394)(7573394)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [803](https://10.10.124.134/job/AR-build-pipeline/803/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7579754)(7579754)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7579754)(7579754)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationTopAppBar.kt [hunk 1]
```diff
@@ -194,6 +180,24 @@ private fun ConversationScreenTopAppBarContent(
     )
 }
 
+@Composable
+private fun VerificationIcon(protocolInfo: Conversation.ProtocolInfo?) {
+    if (protocolInfo?.verificationStatus != Conversation.VerificationStatus.VERIFIED) return
```
Comment (mchenani) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationTopAppBar.kt]:
```suggestion
    if (protocolInfo?.verificationStatus == Conversation.VerificationStatus.NOT_VERIFIED) return
```

**Review by mchenani — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationTopAppBar.kt (hunk 1 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationTopAppBar.kt]: do not agree. If the conversation status is Degraded we shouldn't display any icon just like it's not verified ([design](https://www.figma.com/file/gSmh7BRyq9SGKkyOLSbOdM/E2E-Identity?type=design&node-id=1-8174&mode=design&t=6EnAtJZfA9c1K9Ez-4) )

**Comment by AndroidBob** [comment]
Build [913](https://10.10.124.134/job/AR-build-pipeline/913/) **failed**.

**Comment by AndroidBob** [comment]
Build [937](https://10.10.124.134/job/AR-build-pipeline/937/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7666581)(7666581)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7666581)(7666581)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [999](https://10.10.124.134/job/AR-build-pipeline/999/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7676737)(7676737)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7676737)(7676737)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1025](https://10.10.124.134/job/AR-build-pipeline/1025/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7685333)(7685333)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7685333)(7685333)-staging-compat.apk)

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1038](https://10.10.124.134/job/AR-build-pipeline/1038/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7692242)(7692242)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7692242)(7692242)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1046](https://10.10.124.134/job/AR-build-pipeline/1046/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7693238)(7693238)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7693238)(7693238)-staging-compat.apk)

**Pull request merged**